src/plugins/query_enhancements/server/routes/query_assist/agents.ts

```diff
@@ -37,11 +37,13 @@ export const getAgentIdByConfig = async (
       method: 'GET',
       path: `${URI.ML}/config/${configName}`,
     })) as ApiResponse<{ type: string; configuration: { agent_id?: string } }>;
-
-    if (!response || response.body.configuration.agent_id === undefined) {
+    if (
+      !response ||
+      !(response.body.ml_configuration?.agent_id || response.body.configuration?.agent_id)
```

Comment:
do we need to support both keys? i think we do because query assist for MDS clusters can have older version of ml-commons

might be better to add a test to ensure both `configuration` and `ml_configuration` are supported. currently it only tests `configuration`
https://github.com/opensearch-project/OpenSearch-Dashboards/blob/9f3bbe8be6daf23f2c0577035abecb3686b01c06/src/plugins/query_enhancements/server/routes/query_assist/agents.test.ts#L41